### ISSUE TITLE
Windows VSCode do not show error UI. Im new to swift

### ISSUE THREAD

**Issue by wolonge95** [swiftlang/vscode-swift#1226]
error code like below, vscode editor do not show error annotations, seems like nothing wrong, and not show in Problems view. 
<img width="320" alt="image" src="https://github.com/user-attachments/assets/7d6dada6-b606-4dca-9369-eccbc49daf35">

But have this in Output view :
<img width="770" alt="image" src="https://github.com/user-attachments/assets/87a82cc8-82e4-447e-8e2e-9844d4edc3b7">

Im new to swift. Want to try it. So I install swift follow the instructions on swift.org . Im on windows 11. 

I can run "Hello world" in terminal, so I think everything is ok. But vscode spent a lot of time, it does not work as expected.

**Expected behavior**
Error shows in editor UI, and Problems view.

**Environment**
 - OS: Windows 11 26100.2454
 - Swift version 
     Swift version 6.0.2 (swift-6.0.2-RELEASE)
     Target: x86_64-unknown-windows-msvc
 - Visual Studio Code version: 1.95.3
 - vscode-swift version: 1.11.3 (swift extension from Swift Server Work Group)

**Additional context**
A few hours ago, I installed everything.

**Comment by wolonge95** [comment]
and errors report twice. I do not know it's right or wrong. 
<img width="700" alt="image" src="https://github.com/user-attachments/assets/9ca66c27-85fe-43d2-a0ab-60cadd0bbb6d">

**Comment by plemarquand** [comment]
Hi, are you building your code from the command line or from within VS Code? You may have to try and build your code once first from within VS Code using either `Debug swift-executable` or `Release swift-executable` in the drop down menu with the green play button on the Run and Debug navigator panel.

If that doesn't work and you're still not seeing any problems in the Problems tab could you run `> Swift: Capture VS Code Swift Diagnostic Bundle`, choose `Capture Full Diagnostics`, zip up the produced files and then attach them to this issue? Thanks!

**Comment by wolonge95** [comment]
`Debug swift-executable` can show error in editor and Problems tab, but it is not real time. 

When I modify the code, the error will not disappear. 

[vscode-diagnostics-2024-11-29T20-06-14+08-00.zip](https://github.com/user-attachments/files/17958918/vscode-diagnostics-2024-11-29T20-06-14%2B08-00.zip)

**Comment by plemarquand** [comment]
In your VS Code User `settings.json` (`> Preferences: Open User Settings (JSON)`) Could you try setting your `"swift.diagnosticsStyle"` to `"swift"` instead of `"llvm"`?

**Comment by wolonge95** [comment]
YES. I try it.  Does not work.

By the way, it can show Inline Hint but not error

**Comment by award999** [comment]
Also although "swift.diagnosticsStyle" can be set to "swift" or "llvm", the "llvm" one is better so we can capture related issues

**Comment by wolonge95** [comment]
yes, task outputs problems. 

And inline hint works, code completion works, api doc works.  Just not show error.

**Comment by award999** [comment]
@wolonge95 nothing is jumping out. It seems like you should be able to get diagnostics from the language server or tasks. Can you double check you haven't modified settings like `outline.problems.enabled` or `problems.visibility` which would prevent diagnostics from being shown

**Comment by wolonge95** [comment]
Yeah, I do not change anything. My VSCode works for other languages. settings relate swift:
```
   "lldb.library": "C:\\Users\\wolonge\\AppData\\Local\\Programs\\Swift\\Toolchains\\6.0.2+Asserts\\usr\\bin\\liblldb.dll",
    "lldb.launch.expressions": "native",
    "swift.diagnosticsStyle": "swift",    // or llvm not work
```

**Comment by Star-Lord-PHB** [comment]
Have the same problem here on Windows 11. The editor inline error can be shown after manually running the build all task, but they are not updated in real time. Even after fixing the error, the inline error messages are still there and I have to manually run the build task again

![Screenshot 2024-12-04 at 16 14 59](https://github.com/user-attachments/assets/0c7b0f29-e153-497e-a38d-ca1e5ffb4b08)

After manually running build all task

![Screenshot 2024-12-04 at 16 15 29](https://github.com/user-attachments/assets/628b0283-751f-4b71-8360-9410e0be7763)

After fixing the error

![Screenshot 2024-12-04 at 16 15 38](https://github.com/user-attachments/assets/b0131f52-7e3d-46cd-b9d5-811e9c277b55)

Using the background compilation can be a workaround, but I don't really like it.

**Comment by award999** [comment]
@wolonge95 @Star-Lord-PHB I was able to reproduce. Debugging on windows, the extension was not getting diagnostics from the language server, raised https://github.com/swiftlang/sourcekit-lsp/issues/1855

**Comment by rbenegal** [comment]
Seems like https://github.com/swiftlang/sourcekit-lsp/issues/1855 has been fixed, so possibly things are working now. I can look into this again.

**Comment by rbenegal** [comment]
Confirmed that this issue has been fixed in recent versions of the toolchain. Checked with the main branch version of `swift-DEVELOPMENT-SNAPSHOT-2025-01-04-a-windows10` on Windows 11:

```
>ver
Microsoft Windows [Version 10.0.26100.2605]

>swiftc -version
Swift version 6.2-dev (LLVM 06a77c5cc0ff511, Swift 062061e37a8851b)
Target: x86_64-unknown-windows-msvc
```

<img width="419" alt="Screenshot 2025-01-09 at 10 43 42 AM" src="https://github.com/user-attachments/assets/3c19ab20-7b11-4194-99a5-f59389274f5a" />


For now (as of release 6.0.3) a development snapshot from https://www.swift.org/install/windows/#development-snapshots would be needed to have diagnostic error and warning messages. You may have to uninstall the existing version first to install the new version. Future release versions should work out of the box.

**Comment by plemarquand** [comment]
Great! Thanks for checking this out @rbenegal.
